mockturtle/app/commands.rb

```diff
@@ -0,0 +1,122 @@
+require "ircinch"
+require "json"
+require "nokogiri"
+require "open-uri"
+require "time"
+
+class Seen
+  class SeenStruct < Struct.new(:who, :where, :time)
```

Comment:
Since these aren't modified after creation, recent versions of Ruby (which we're running) have a [`Data`](https://docs.ruby-lang.org/en/3.2/Data.html) class that's immutable.

I'd also prefer to just be `Seen` without the hungarian notation. Ruby is very timey-wimey about types.